src/grammar/parser.y

```diff
@@ -1665,11 +1665,21 @@ ClassOrInterfaceType: TypeDeclSpecifier
 // Actually
 // TypeDeclSpecifier: TypeName | ClassOrInterfaceType . Identifier
 // TypeName:          Identifier | TypeName . Identifier
-TypeDeclSpecifier: QualifiedIdentifier
-                 | ClassOrInterfaceType DOT IDENTIFIER 
-                   { 
+TypeDeclSpecifier:
+                 IDENTIFIER
+                 | Annotations_opt IDENTIFIER
+                   {
+                     $$ = $2;
+                   }
+                 | ClassOrInterfaceType DOT IDENTIFIER
+                   {
                      $$ = $1.getName() + '.' + $3;
-                   };
+                   }
+                 | ClassOrInterfaceType DOT Annotations_opt IDENTIFIER
```

Comment:
The specs don't allow an optional annotation here.
I've tried hard to get this file closer to the Java language specification, this line just doesn't match with the comments above...

Reply:
That's there to cover this situation (out of the specs):

> PackageName . {Annotation} TypeIdentifier [TypeArguments]
ClassOrInterfaceType . {Annotation} TypeIdentifier [TypeArguments]